core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -1686,12 +1689,23 @@ private void processBind(long lo, long msgLimit, @Transient SqlCompiler compiler
         validateParameterCounts(parameterFormatCount, parameterValueCount, parsePhaseBindVariableCount);
 
         lo += Short.BYTES;
-        if (parameterValueCount > 0) {
-            if (this.parsePhaseBindVariableCount == parameterValueCount) {
-                lo = bindValuesUsingSetters(lo, msgLimit, parameterValueCount);
-            } else {
-                lo = bindValuesAsStrings(lo, msgLimit, parameterValueCount);
+
+        try {
+            if (parameterValueCount > 0) {
+                if (this.parsePhaseBindVariableCount == parameterValueCount) {
+                    lo = bindValuesUsingSetters(lo, msgLimit, parameterValueCount);
+                } else {
+                    lo = bindValuesAsStrings(lo, msgLimit, parameterValueCount);
+                }
+            }
+        } catch (SqlException e) {
+            //if statement is named or portal then we've to add it to typesAndSelect
+            if (typesAndSelect != null) {
+                Misc.free(typesAndSelect);
+                typesAndSelect = null;
+                //alternative would be to add typesAndSelect to cache  
```

Comment:
nit: we can probably remove this comment since it makes sense only during review